File: src/c/magma/io/Path.c
import magma.collect.Set;
import magma.option.Option;
import magma.result.Result;
import magma.stream.Stream;
struct Path {Stream<Path> streamNames();boolean exists();Result<Set<Path>, Error> walk();Result<String, Error> readString();Option<Error> writeString(String output);Option<Error> createDirectories();Path relativize(Path child);Option<Path> findParent();Path resolve(String segment);boolean isRegularFile();Path findFileName();
}

File: src/java/magma/api/io/Path.java
package magma.api.io;

import magma.api.option.Option;
import magma.api.result.Result;
import magma.api.stream.Stream;
import magma.java.JavaSet;

import java.io.IOException;

public interface Path {
    boolean isExists();

    Stream<Path> stream();

    Option<IOException> writeString(String output);

    Option<IOException> createAsDirectories();

    Result<String, IOException> readStrings();

    Result<JavaSet<Path>, IOException> walkWrapped();

    java.nio.file.Path unwrap();

    Path relativize(Path child);

    Option<Path> findParent();

    int getNameCount();

    Option<Path> getName(int index);

    String format();

    Path getFileName();

    Path resolvePath(Path child);

    Path resolveChild(String child);
}
